dsgrid/utils/spark.py

```diff
@@ -400,6 +406,30 @@ def sql_from_sqlalchemy(query):
     return sql(str(query).replace('"', ""))
 
 
+@contextmanager
+def custom_spark_conf(conf):
+    """Apply a custom Spark configuration for the duration of a code block.
+
+    Parameters
+    ----------
+    conf : dict
+        Key-value pairs to set on the spark configuration.
+
+    """
+    spark = _get_spark_session()
+    orig_settings = {}
+
+    try:
+        for key, val in conf.items():
+            orig_settings[key] = spark.conf.get(key)
+            spark.conf.set(key, val)
+            logger.info("Set %s=%s while running", key, val)
+        yield
+    finally:
+        for key, val in orig_settings.items():
```

Comment:
Why is the orig setting being set back here? If try fails, orig_settings will not be complete?

Reply:
Cases:
1. Suppose the first new setting is correct but the second one is incorrect, and an exception is raised. We need to revert the first change.
2. The code block wrapped by this decorator will run when `yield` is called. We need to revert all changes afterwards, regardless of whether it completes successfully or raises an exception.